src/queries/useApplicationUpdateMutation.ts

```diff
@@ -0,0 +1,23 @@
+import { useMutation } from '@tanstack/react-query';
+import { Application } from '~/@types/common';
+import { MutationOptions } from '~/@types/requests';
+import useDatabase from '~/hooks/useDatabase';
+
+interface UpdateVariables {
+  id: number;
+  data: Partial<Application>;
+}
+
+const useApplicationUpdateMutation = (
+  options?: MutationOptions<Application, unknown, UpdateVariables>,
+) => {
+  const { applications } = useDatabase();
+
+  return useMutation<Application, unknown, UpdateVariables>({
+    mutationKey: ['APPLICATION_UPDATE'],
```

Comment:
Don't we need applicationId in mutationKey? Not sure what do we use this key for

Reply:
I don't think so, because this id is passed directly to request function.
Maybe if we passed it through props to this hook, we would have to add it here. (but also not sure if it is really necessary for mutations)